system-tests-new/psyclone/test_psyclone.py

```diff
@@ -0,0 +1,174 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+import filecmp
+import shutil
+from os import unlink
+from pathlib import Path
+from typing import Tuple
+from unittest import mock
+
+import pytest
+
+from fab.build_config import BuildConfig
+from fab.parse.x90 import X90Analyser, AnalysedX90
+from fab.steps.find_source_files import FindSourceFiles
+from fab.steps.grab.folder import GrabFolder
+from fab.steps.preprocess import fortran_preprocessor
+from fab.steps.psyclone import make_parsable_x90, Psyclone, psyclone_preprocessor, tool_available, MpPayload
+from fab.util import file_checksum
+
+SAMPLE_KERNEL = Path(__file__).parent / 'kernel.f90'
```

Comment:
For extra test coverage in the future we might want to try an algorithm which calls more than one (none built-in) kernel.

Reply:
Agreed. Added a todo for that so it's not forgotten.